src/backend/app.py

```diff
@@ -98,12 +98,18 @@ def mainpage_get(mainpage_obj: MainPage, args: MultiDict):
         apts = []
         price_sort = 0
         rating_sort = 0
+        apt_id = -1
         if param.price_sort is not None:
             price_sort = param.price_sort
 
         if param.rating_sort is not None:
             rating_sort = param.rating_sort
-        apts = mainpage_obj.populate_apartments(param.num_apts, price_sort, rating_sort)
+
+        if param.apt_id is not None:

```

Comment:
This is only a suggestion, but it's nicer to me when the nested loops are in helper functions. I know it only looks pretty long because of all the lining preventing **>** 100 characters or something.

Reply:
I planned to separate these ifs but forgot lol, will get to fixing it.